lisa/microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py

```diff
@@ -1338,6 +1426,707 @@ def _set_block_size_env_var(self, testcase: str) -> None:
+
+    def _run_anchor_gate(self) -> None:
+        """
+        Anchor gate: 5-8s CPU/mem warmup to validate system stability.
+
+        Validates against EWMA baseline (±5%). Retries once on failure.
+        Uses exponential weighted moving average (alpha=0.3) for stability.
+        """
+        # 5-8s CPU/mem anchor using dd
+        start = time.time()
+        self.node.execute(
+            "dd if=/dev/zero of=/dev/null bs=1M count=2048 status=none",
+            shell=True,
+            timeout=15,
+        )
+        elapsed = time.time() - start
+
+        # Calculate throughput (GB/s) - 2048 MB = 2.048 GB
+        throughput = 2.048 / elapsed
+
+        # EWMA validation (skip first run to establish baseline)
+        if self._anchor_ewma_count > 0:
+            deviation = abs(throughput - self._anchor_ewma) / self._anchor_ewma
+
+            if deviation > self.ANCHOR_DEVIATION_THRESHOLD:
+                self._log.debug(
+                    f"Anchor gate FAILED: {deviation * 100:.1f}% deviation "
+                    f"(expected: {self._anchor_ewma:.2f} GB/s, "
+                    f"got: {throughput:.2f} GB/s)"
+                )
+
+                # Retry once
+                self._log.debug("Retrying anchor gate...")
+                time.sleep(5)
+
+                start = time.time()
+                self.node.execute(
+                    "dd if=/dev/zero of=/dev/null bs=1M count=2048 status=none",
+                    shell=True,
+                    timeout=15,
+                )
+                elapsed = time.time() - start
+                throughput = 2.048 / elapsed
+
+                deviation = abs(throughput - self._anchor_ewma) / self._anchor_ewma
+
+                if deviation > self.ANCHOR_DEVIATION_THRESHOLD:
+                    self._log.debug(
+                        f"Anchor gate FAILED on retry: "
+                        f"{deviation * 100:.1f}% deviation"
+                    )
+                    self._log_thermal_health("anchor_failure")
+                    # Don't fail the suite, but log prominently
+                else:
+                    self._log.debug(
+                        f"Anchor gate PASSED on retry: {throughput:.2f} GB/s"
+                    )
+            else:
+                self._log.debug(f"Anchor gate PASSED: {deviation * 100:.1f}% deviation")
+        else:
+            self._log.debug("Anchor baseline established")
+
+        # Update EWMA (alpha=0.3 for responsiveness)
+        alpha = self.ANCHOR_EWMA_ALPHA
+        if self._anchor_ewma_count == 0:
+            self._anchor_ewma = throughput
+        else:
+            self._anchor_ewma = alpha * throughput + (1 - alpha) * self._anchor_ewma
+
+        self._anchor_ewma_count += 1
+
+        # Log anchor measurements for correlation across runs
+        self._log.info(
+            f"[ANCHOR_METRICS] measurement={throughput:.3f} GB/s, "
+            f"ewma={self._anchor_ewma:.3f} GB/s, count={self._anchor_ewma_count}, "
```

Comment:
Is this data even reliable given that this was calculated with only 2 data points?

Reply:
intent of the anchor is advisory, not a benchmark or gate,. ok we can remove it